embassy-nrf/src/pwm.rs

```diff
@@ -117,55 +121,70 @@ impl<'d, T: Instance> SequencePwm<'d, T> {
         r.countertop
             .write(|w| unsafe { w.countertop().bits(config.max_duty) });
 
-        r.seq0.refresh.write(|w| unsafe { w.bits(config.refresh) });
-        r.seq0
-            .enddelay
-            .write(|w| unsafe { w.bits(config.end_delay) });
-
-        r.seq1.refresh.write(|w| unsafe { w.bits(config.refresh) });
-        r.seq1
-            .enddelay
-            .write(|w| unsafe { w.bits(config.end_delay) });
-
         Ok(Self {
             phantom: PhantomData,
             ch0: ch0.degrade_optional(),
             ch1: ch1.degrade_optional(),
             ch2: ch2.degrade_optional(),
             ch3: ch3.degrade_optional(),
+            sequence0: None,
+            sequence1: None,
         })
     }
 
-    /// Start or restart playback
+    /// Start or restart playback. Takes at least one sequence along with its
+    /// configuration. Optionally takes a second sequence and its configuration.
+    /// In the case where no second sequence is provided then the first sequence
+    /// is used. The sequence mode applies to both sequences combined as one.
     #[inline(always)]
-    pub fn start(&mut self, sequence: &'d [u16], times: SequenceMode) -> Result<(), Error> {
-        slice_in_ram_or(sequence, Error::DMABufferNotInDataMemory)?;
-
-        if sequence.len() > 32767 {
+    pub fn start(
+        &mut self,
+        sequence0: Sequence<'d>,
+        sequence1: Option<Sequence<'d>>,
+        times: SequenceMode,
+    ) -> Result<(), Error> {
+        let alt_sequence = sequence1.as_ref().unwrap_or(&sequence0);
+
+        slice_in_ram_or(sequence0.words, Error::DMABufferNotInDataMemory)?;
+        slice_in_ram_or(alt_sequence.words, Error::DMABufferNotInDataMemory)?;
```

Comment:
Because the `Sequence` struct now retains the word buffer as mutable, do we need this check? I'm thinking not...

Reply:
why is it mutable now? `&'d [u16]` should be enough, we just need it stays alive for the duration of the DMA, we don't need to write to it.

Reply:
It is mutable so that we can subsequently write into it. You’ll see that the stop method transfers ownership back. The example illustrates the use case where we are updating the buffer so that the led pulsates. 

If we cant update the buffer then we’d need a number of stack-allocated buffers for each frame of the pulsating effect. This would be expensive on memory, particularly when more than one LED needs to be addressed.

Reply:
Commit 2112b83e9d328bd662156a167e721ab28bffc6f7
d76cd5ceaf5140c48ef97180beae156c0c0e07c8 now owns the sequence buffers. This approach owns the sequence buffers which, while introducing an extra move, eliminates the need to guard the lifetime of the sequence buffer. Given ownership, the buffer will be retained until the PWM sequence task is stopped. WDYT?

Reply:
I've now reverted the owning of buffers - too much copying for comfort. So, we're back to passing around &'d mut [16]`, which I think is fine given the opportunity to subsequently modify it. I'm pretty happy with things now.

Reply:
Sequence buffer is no longer mutable.